<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Brian Maitner" />

<meta name="date" content="2023-01-05" />

<title>BIEN R package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>







<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">BIEN R package</h1>
<h4 class="author">Brian Maitner</h4>
<h4 class="date">2023-01-05</h4>



<p>The Botanical Information and Ecology Network(BIEN) R package
provides access to the BIEN database as well as useful tools for working
with the BIEN data.</p>
<p>##Data Usage Agreement Please read the data usage agreement,
available at: <a href="https://bien.nceas.ucsb.edu/bien/biendata/data-use-policy/" class="uri">https://bien.nceas.ucsb.edu/bien/biendata/data-use-policy/</a></p>
<p>##Data Usage: Warnings and Caveats</p>
<p>Please be aware of quality and appropriateness of BIEN data for the
analyses that you may wish to perform or information that you would like
to obtain</p>
<p>Versioning - data available via BIENdata.org and the the BIEN R
package (BIEN) are being updated and improved. This is an active
collaborative group who is aiming to continue to improve the quality of
botanical information and data. As a result, users should take note of
the version number (using the function BIEN_metadata_database_version()
) and be sure to check back periodically for updates. Checking for new
results to a query can be done using the function
BIEN_metadata_match_data().</p>
<p>Data quality - All data are presented ‘as is’. Default BIEN downloads
return all data associated with a given taxonomic name string. There may
be additional issues associated with a given name string. Please be
aware that data returned may also include data whose geographic
coordinates and cultivated status are uncertain or unverifiable.</p>
<p>Hybrids - Hybrid species are assigned the specific epithet “x” by the
TNRS. To view the full name information for these occurrences, set
“all.taxonomy = TRUE”.</p>
<p>##Database connection issues Some institution and computer programs
(e.g. some antivirus programs) block the SQL connections that this
package relies on. While we are exploring ways around this issue, at
present the simplest method is to use the package on a computer/network
that doesn’t block SQL connections.</p>
<p>##Function Names Function names in the BIEN package follow a naming
convention which was designed to make locating the desired function
easier:</p>
<ul>
<li>All function names begin with the prefix “BIEN_”</li>
<li>Functions to download occurrence records begin with the prefix
“BIEN_occurrence_”</li>
<li>Functions to download range maps begin with the prefix
“BIEN_ranges_”</li>
<li>Functions to download trait data begin with the prefix
“BIEN_trait_”</li>
<li>Functions to download species lists (or other lists) begin with the
prefix “BIEN_list_”</li>
<li>Functions to download plot data begin with the prefix
“BIEN_plot_”</li>
<li>Functions to download stem data begin with the prefix
“BIEN_stem_”</li>
<li>Functions to download taxonomic information begin with the prefix
“BIEN_taxonomy_”</li>
<li>Functions to download phylogenies begin with the prefix
“BIEN_phylogeny_”</li>
<li>Functions to access metadata begin with the prefix
“BIEN_metadata_”</li>
</ul>
<p>##Function Directory For full information on each function, see the
associated help file.</p>
<p>###BIEN_metadata functions *
<code>BIEN_metadata_database_version</code> Returns the BIEN database
version number and release date * <code>BIEN_metadata_match_data</code>
Compares the stored results of two (old vs new) queries to check for
additions or deletions. * <code>BIEN_metadata_citation</code> Generates
information needed to cite data downloaded from BIEN. *
<code>BIEN_metadata_list_political_names</code> Provides a list of
political division names used by BIEN.</p>
<p>###BIEN_list functions * <code>BIEN_list_all</code> Lists all species
present in the BIEN database * <code>BIEN_list_country</code> Produces
species lists by country * <code>BIEN_list_county</code> Produces
species lists by county * <code>BIEN_list_state</code> Produces species
lists by state/province * <code>BIEN_list_sf</code> Produces species
lists using an sf object</p>
<p>###BIEN_occurrence functions * <code>BIEN_occurrence_box</code>
Returns all occurrence records within a GIS bounding box *
<code>BIEN_occurrence_country</code> Returns all occurrence records
within a given country * <code>BIEN_occurrence_family</code> Returns all
occurrence records for a specified family *
<code>BIEN_occurrence_genus</code> Returns all occurrence records for a
specified genus * <code>BIEN_occurrence_species</code> Returns all
occurrence records for a specified species *
<code>BIEN_occurrence_state</code> Returns all occurrences records
within a given state/province *
<code>BIEN_occurrence_occurrences_per_species</code> Returns data on the
number of occurrence records for species in the BIEN database *
<code>BIEN_occurrence_sf</code> Returns occurrence records falling
within the boundaries of an sf object.</p>
<p>###BIEN_ranges functions * <code>BIEN_ranges_genus</code> Downloads
range maps for all the species within a given genus/genera *
<code>BIEN_ranges_species</code> Downloads range maps for given species
* <code>BIEN_ranges_box</code> Downloads range maps intersecting a given
GIS bounding box * <code>BIEN_ranges_list</code> Lists available range
maps * <code>BIEN_ranges_sf</code> Downloads range maps intersecting a
user-supplied sf object.</p>
<p>###BIEN_trait functions * <code>BIEN_trait_family</code> Returns all
available trait data for given families * <code>BIEN_trait_genus</code>
Returns all available trait data for given genera *
<code>BIEN_trait_list</code> Lists all available types of trait *
<code>BIEN_trait_mean</code> Estimates species mean trait values using
Genus or Family level data where needed *
<code>BIEN_trait_species</code> Returns all available trait data for
given species * <code>BIEN_trait_trait</code> Returns all available
trait data for given trait(s) * <code>BIEN_trait_traitbyfamily</code>
Returns specific trait data for given family *
<code>BIEN_trait_traitbygenus</code> Returns specific trait data for
given genus * <code>BIEN_trait_traitbyspecies</code> Returns specific
trait data for given species *
<code>BIEN_trait_traits_per_species</code> Returns data on the number of
trait observations for each trait for each species in the BIEN
database</p>
<p>###BIEN_plot functions * <code>BIEN_plot_datasource</code> Downloads
all plot data from a given datasource. *
<code>BIEN_plot_list_datasource</code> List all datasources in the BIEN
database. * <code>BIEN_plot_country</code> Downloads all plot data from
specified countries. * <code>BIEN_plot_state</code> Downloads all plot
data from specified states/provinces. *
<code>BIEN_plot_list_sampling_protocols</code> List all available
sampling protocols. * <code>BIEN_plot_sampling_protocol</code> Downloads
all plot data using a specified sampling protocol.</p>
<p>###BIEN_stem functions * <code>BIEN_stem_species</code> Downloads all
stem data for given species. * <code>BIEN_stem_genus</code> Downloads
all stem data for given genera. * <code>BIEN_stem_family</code>
Downloads all stem data for given families.</p>
<p>###BIEN_taxonomy functions * <code>BIEN_taxonomy_species</code>
Downloads all taxonomic information for given species. *
<code>BIEN_taxonomy_genus</code> Downloads all taxonomic information for
given genera. * <code>BIEN_taxonomy_family</code> Downloads all
taxonomic information for given families.</p>
<p>###BIEN_phylogeny functions * <code>BIEN_phylogeny_complete</code>
Downloads a specified number of the BIEN phylogeny replicates. *
<code>BIEN_phylogeny_conservative</code> Downloads the conservative BIEN
phylogeny, which only includes species with molecular data
available.</p>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
